From 36c872a6260a472e2f0fc011fa2ddc5bfe8b3aff Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 16 Feb 2021 23:57:53 +0200
Subject: [PATCH] [SW-29740] gaudi: security: move sm to use rr28-31

Since we are requested to use additional RRs for protecting more LBW
areas, move the SYNC_MNGR protection (which is PCIe only) to the last
PCIe RRs, so those ranges used till now (2-5) will be free for
additional LBW use, but still have a PCIe HBW dedicated registers
contiguous.
So the new PCIe RRs dedication:
LBW: 0-15, 28-31
HBW: 16-17

Change-Id: Ib4a47b1cc9ba7eb3bc26dddc18ff2c24e1094789
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index a5c7c24..45ed860 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1189,14 +1189,6 @@
 	max = mmDMA_IF_W_S_BASE;
 	preboot_secure_lbw_rr(1, min, max);
 
-	/* SYNC_MNGR: RR_2 up to RR_5 */
-	for (i = 0; i < 4; i++) {
-		base = mmSYNC_MNGR_GLBL_W_S_BASE + SYNC_MNGR_DIST * i;
-		min = base - 1;
-		max = base + SYNC_MNGR_SIZE;
-		preboot_secure_pcie_rr(2 + i, min, max);
-	}
-
 	/* HBM_CFG up to GIC_L: RR_6 */
 	min = mmHBM0_BASE - 1;
 	max = HL_CFG_BASE + mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR;
@@ -1251,6 +1243,14 @@
 	min = HL_CFG_BASE + mmPSOC_ETR_AXICTL - 1;
 	max = HL_CFG_BASE + mmPSOC_ETR_AXICTL + sizeof(u32);
 	preboot_secure_lbw_rr(15, min, max);
+
+	/* SYNC_MNGR: PCIe only! PCIe RR_28 up to PCIe RR_31 */
+	for (i = 0; i < 4; i++) {
+		base = mmSYNC_MNGR_GLBL_W_S_BASE + SYNC_MNGR_DIST * i;
+		min = base - 1;
+		max = base + SYNC_MNGR_SIZE;
+		preboot_secure_pcie_rr(28 + i, min, max);
+	}
 }
 
 static void preboot_secure_hbw_rr(int rr, u64 base, u64 mask)
@@ -1385,7 +1385,8 @@
 	/*
 	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
 	 * STM, SPI Flash, Scratchpad SRAM, PCIe FW SRAM.
-	 * Assign the PCIe IF RR_16 (0-15 are allocated for LBW) for this job.
+	 * Assign the PCIe IF RR_16 (0-15 and 28-31 are allocated for LBW) for
+	 * this job.
 	 * NOTE: The PCIe RRs use min/max concept and HIT is: min < hit < max
 	 */
 	preboot_secure_pcie_rr(16, mmSTM_BASE - 1, mmRAZWI_BASE);
